File: notifications/src/main/java/com/datahub/notification/NotificationTracking.java
package com.datahub.notification;

import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class NotificationTracking {

  public static final String NOTIFICATION_QUERY_PARAM_TYPE = "notification_type";
  public static final String NOTIFICATION_QUERY_PARAM_ID = "notification_id";
  public static final String NOTIFICATION_QUERY_PARAM_CHANNEL = "notification_channel";

  public enum NotificationChannel {
    SLACK("slack");

    private final String value;

    NotificationChannel(String value) {
      this.value = value;
    }

    public String getValue() {
      return value;
    }
  }

  public enum NotificationType {
    ASSERTION("assertion"),
    INCIDENT("incident");

    private final String value;

    NotificationType(String value) {
      this.value = value;
    }

    public String getValue() {
      return value;
    }
  }
}

File: datahub-integrations-service/src/datahub_integrations/notifications/notification_tracking.py
from enum import Enum

NOTIFICATION_QUERY_PARAM_TYPE = "notification_type"
NOTIFICATION_QUERY_PARAM_ID = "notification_id"
NOTIFICATION_QUERY_PARAM_CHANNEL = "notification_channel"


class NotificationChannel(str, Enum):
    SLACK = "slack"
    EMAIL = "email"
    TEAMS = "teams"


class NotificationType(str, Enum):
    ASSERTION = "assertion"
    INCIDENT = "incident"
